experimental/PropertyDDS/packages/property-proxy/src/arrayProxyHandler.ts

```diff
@@ -155,9 +155,9 @@ export const arrayProxyHandler = {
      * @param key The name of the property/function that is to be accessed.
      * @return if the key is part of the {@link external:ArrayProperty ArrayProperty}, otherwise false.
      */
-    has: (target: ComponentArray, key: string | number | typeof proxySymbol) =>
+    has: (target: ComponentArray, key: PropertyKey) =>
         key === "swap" || key in [] || key === proxySymbol ||
-        (key >= 0 && key < target.getProperty().getLength()),
+        (key as number >= 0 && key as number < target.getProperty().getLength()),
```

Comment:
I'd rather not see us adding more `as`, is there another way to go here?

Reply:
Good point. This code ended up that way because it's kind of broken, but I have now fixed it properly. I reworked this to avoid `as`, point out the typescript bug thats relevant, and handle non integer keys correctly.
I didn't really want to include bug fixes in this, but I'm not going to leave that fix out.